staging/src/k8s.io/apiserver/pkg/endpoints/handlers/responsewriters/writers.go

```diff
@@ -89,8 +89,7 @@ func StreamObject(statusCode int, gv schema.GroupVersion, s runtime.NegotiatedSe
 // SerializeObject renders an object in the content type negotiated by the client using the provided encoder.
 // The context is optional and can be nil. This method will perform optional content compression if requested by
 // a client and the feature gate for APIResponseCompression is enabled.
-func SerializeObject(mediaType string, encoder runtime.Encoder, hw http.ResponseWriter, req *http.Request, statusCode int, object runtime.Object) {
-	ctx := req.Context()
+func SerializeObject(ctx context.Context, mediaType string, encoder runtime.Encoder, hw http.ResponseWriter, req *http.Request, statusCode int, object runtime.Object) {
```

Comment:
It feels a bit weird that we're passing a ctx object to the function that should be encoded in the request context. Is the issue that `request.Context()` is not initialized so the context is created as context.Background() too late and that's why the spans don't align? Would it be better to the request context via sending `req.WithContext(...)` instead of a regular `req` object? That way the function signatures don't have to change

Reply:
Eg on the last line in get.go: 

```
transformResponseObject(scope, req.WithContext(ctx), w, http.StatusOK, outputMediaType, result)
```

Reply:
Thanks a lot for your nice suggestion!
I would test base on your input.

Reply:
> It feels a bit weird that we're passing a ctx object to the function that should be encoded in the request context. Is the issue that `request.Context()` is not initialized so the context is created as context.Background() too late and that's why the spans don't align? Would it be better to the request context via sending `req.WithContext(...)` instead of a regular `req` object? That way the function signatures don't have to change

@Jefftree 

I have made the modifications based on your comments, ensuring that the function signatures remain unchanged. From the trace flow below, including the fix with the API server, it is working as expected. Thanks!


![image](https://github.com/kubernetes/kubernetes/assets/6135084/95f4699e-2f19-4da5-af61-eb49af168c33)

Reply:
![image](https://github.com/kubernetes/kubernetes/assets/6135084/90716674-0f75-4042-ac21-e8735f06f035)